vaadin-card-flow-parent/vaadin-card-flow/src/main/java/com/vaadin/flow/component/card/Card.java

```diff
@@ -32,6 +48,337 @@
+     * {@link #setSubtitle(Component) subtitle} components, and will be
+     * displayed instead.
+     * <p>
+     * Passing {@code null} removes the current header component from the card.
+     *
+     * @param header
+     *            the header component, or {@code null} to remove
+     */
+    public void setHeader(Component header) {
+        setSlotContent(HEADER_SLOT_NAME, header);
+    }
+
+    /**
+     * Gets the current header component.
+     *
+     * @return the header component, or {@code null} if none is set
+     */
+    public Component getHeader() {
+        return getSlotContent(HEADER_SLOT_NAME);
+    }
+
+    /**
+     * Sets a component to the header prefix slot, displayed before the header
+     * content.
+     * <p>
+     * Passing {@code null} removes any existing prefix from the header.
+     *
+     * @param headerPrefix
+     *            the header prefix component, or {@code null} to remove
+     */
+    public void setHeaderPrefix(Component headerPrefix) {
+        setSlotContent(HEADER_PREFIX_SLOT_NAME, headerPrefix);
+    }
+
+    /**
+     * Gets the current header prefix component.
+     *
+     * @return the header prefix component, or {@code null} if none is set
+     */
+    public Component getHeaderPrefix() {
+        return getSlotContent(HEADER_PREFIX_SLOT_NAME);
+    }
+
+    /**
+     * Sets a component to the header suffix slot, displayed after the header
+     * content. Commonly used for decorative icons or custom suffixes.
+     * <p>
+     * Passing {@code null} removes any existing suffix from the header.
+     *
+     * @param headerSuffix
+     *            the header suffix component, or {@code null} to remove
+     */
+    public void setHeaderSuffix(Component headerSuffix) {
+        setSlotContent(HEADER_SUFFIX_SLOT_NAME, headerSuffix);
+    }
+
+    /**
+     * Gets the current header suffix component.
+     *
+     * @return the header suffix component, or {@code null} if none is set
+     */
+    public Component getHeaderSuffix() {
+        return getSlotContent(HEADER_SUFFIX_SLOT_NAME);
+    }
+
+    /**
+     * Adds components to the card's footer slot.
+     *
+     * @param footerComponent
+     *            the components to add into the footer
+     */
+    public void addToFooter(Component... footerComponent) {
+        Objects.requireNonNull(footerComponent,
+                "Components should not be null");
+        var componentsToAdd = Arrays.stream(footerComponent)
+                .map(component -> Objects.requireNonNull(component,
+                        "Component to add cannot be null"))
+                .toList();
+        componentsToAdd.forEach(component -> SlotUtils.addToSlot(this,
+                FOOTER_SLOT_NAME, component));
+    }
+
+    /**
+     * Gets all components added to the card's footer.
+     *
+     * @return an array of footer components
+     */
+    public Component[] getFooterComponents() {
```

Comment:
Should it also return a Stream like `getChildren` does?

```suggestion
    public Stream<Component> getFooterComponents() {
```

Reply:
I'm not sure about this one. We don't seem to return `Stream` except for when we override `getChildren`. But the use of `List<Component>` instead of `Component[]` would also make sense. What do you think? The original signature in the ticket specifies `Component[]`, that's why I added it like that initially.